### PR TITLE
[CAT-151] v5.0

### PR DESCRIPTION
**Pull request by goatrocks**
## Summary

added addmodelgroupcomponent query and class.

todo: 

- [x] tests
- [ ] fix valid actions 
- [x] redirect existing questionniare classes.
- [x] cleanup docs/examples

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [ ] Jira story/task is put into PR title
- [x] Code has been self-reviewed
- [x] Hard-to-understand areas have been commented
- [x] Documentation has been updated
- [x] My changes generate no new warnings
- [x] Unit tests have been added for base functionality and known edge cases

### CODE REVIEW

File: indico/types/model_group.py [hunk 1]
```diff
@@ -24,3 +27,44 @@ class ModelGroup(BaseType):
...
+    CLASSIFICATION = 1
+    FORM_EXTRACTION = 2
+    OBJECT_DETECTION = 3
+
+class ModelType(Enum):
+    STANDARD = 1
+    FINETUNE = 2
+    OBJECT_DETECTION = 3
+    FORM_EXTRACTION = 4
+    DOCUMENT = 5
```
Comment (meghanhickey) [inline indico/types/model_group.py:43]: We also use `TFIDF_LR` and `TFIDF_GBT` model types for classification. The default classification type is `TFIDF_GBT`

```suggestion
    DOCUMENT = 5
    TFIDF_LR = 6
    TFIDF_GBT = 7
```

File: indico/queries/workflow.py [hunk 2]
```diff
@@ -417,3 +440,154 @@ def requests(self):
...
+         name: str,
+        source_column_id: int,
+        after_component_id: int,
+        labelset_column_id: int = None
+        new_labelset_args: NewLabelsetArguments = None
+        new_questionnaire_args: NewQuestionaireArguments = None
+
+    """
+    query = """
+            mutation addExtractionModel(
```
[the remark this answers is not in the document]
Comment (goatrocks) [reply indico/queries/workflow.py:497]: oh, true!

File: indico/queries/model_groups/model_groups.py [hunk 3]
```diff
@@ -395,3 +360,126 @@ def requests(self):
         yield _ModelGroupPredict(
             model_id=self.model_id, data=self.data, predict_options=self.predict_options
         )
+
+
+class AddModelGroupComponent(GraphQLRequest):
+    """
+    Adds a new model group component to a workflow, optionally with a customized questionnaire. Available on 5.0+ only.
```
Comment (Scott771) [inline indico/queries/model_groups/model_groups.py:367]: I think "component" is going to be a new term for people (honestly I'm not 100% sure what it means)

**Review by Scott771 — APPROVED** [review]

File: indico/queries/model_groups/model_groups.py (hunk 3 above)
Comment (goatrocks) [reply indico/queries/model_groups/model_groups.py:367]: Good point, I might see if there's a better term or phrasing, for now I'll specify it's a step of sorts
